tests/access_control/GranularGuardianAccessControl.t.sol

```diff
@@ -0,0 +1,327 @@
+    vm.startPrank(retryGuardian);
+    _retryTx(ccc);
+    vm.stopPrank();
+  }
+
+  function test_retryTxWhenWrongCaller(
+    address defaultAdmin,
+    address retryGuardian,
+    address solveEmergencyGuardian,
+    address ccc,
+    address retryCaller
+  ) public setUpGranularGuardians(defaultAdmin, retryGuardian, solveEmergencyGuardian, ccc) {
+    vm.assume(retryCaller != retryGuardian);
+    hoax(retryCaller);
+    vm.expectRevert(
+      bytes(
+        string.concat(
+          'AccessControl: account 0x',
+          TestUtils.toAsciiString(retryCaller),
+          ' is missing role 0xc448b9502bbdf9850cc39823b6ea40cfe96d3ac63008e89edd2b8e98c6cc0af3'
+        )
+      )
+    );
+    _retryTx(ccc);
+  }
+
+  function test_retryEnvelope(
+    address defaultAdmin,
+    address retryGuardian,
+    address solveEmergencyGuardian,
+    address ccc,
+    bytes32 mockEnvId
+  ) public setUpGranularGuardians(defaultAdmin, retryGuardian, solveEmergencyGuardian, ccc) {
+    vm.startPrank(retryGuardian);
+    //    bytes32 mockEnvId = keccak256(abi.encode('mock envelope id'));
+    bytes32 envId = _retryEnvelope(ccc, mockEnvId);
+    assertEq(envId, mockEnvId);
+    vm.stopPrank();
+  }
+
+  function test_retryEnvelopeWhenWrongCaller(
+    address defaultAdmin,
+    address retryGuardian,
+    address solveEmergencyGuardian,
+    address ccc,
+    address retryCaller,
+    bytes32 mockEnvId
+  ) public setUpGranularGuardians(defaultAdmin, retryGuardian, solveEmergencyGuardian, ccc) {
+    vm.assume(retryCaller != retryGuardian);
+    hoax(retryCaller);
+    vm.expectRevert(
+      bytes(
+        string.concat(
+          'AccessControl: account 0x',
+          TestUtils.toAsciiString(retryCaller),
+          ' is missing role 0xc448b9502bbdf9850cc39823b6ea40cfe96d3ac63008e89edd2b8e98c6cc0af3'
+        )
+      )
+    );
+    _retryEnvelope(ccc, mockEnvId);
+  }
+
+  function test_solveEmergency(
+    address defaultAdmin,
+    address retryGuardian,
+    address solveEmergencyGuardian,
+    address ccc
+  ) public setUpGranularGuardians(defaultAdmin, retryGuardian, solveEmergencyGuardian, ccc) {
+    vm.startPrank(solveEmergencyGuardian);
+    _solveEmergency(ccc);
+    vm.stopPrank();
+  }
+
+  function test_solveEmergencyWhenWrongCaller(
+    address defaultAdmin,
+    address retryGuardian,
+    address solveEmergencyGuardian,
+    address ccc,
+    address solveCaller
+  ) public setUpGranularGuardians(defaultAdmin, retryGuardian, solveEmergencyGuardian, ccc) {
+    vm.assume(solveCaller != solveEmergencyGuardian);
+    hoax(solveCaller);
+    vm.expectRevert(
+      bytes(
+        string.concat(
+          'AccessControl: account 0x',
+          TestUtils.toAsciiString(solveCaller),
+          ' is missing role 0xf4cdc679c22cbf47d6de8e836ce79ffdae51f38408dcde3f0645de7634fa607d'
+        )
+      )
+    );
+    _solveEmergency(ccc);
+  }
+
+  function test_updateGuardian(
```

Comment:
this test seems to fail, to replicate increase the number of runs in `foundry.toml` 

counterexample: `args=[0x000000000000000000000000ffFFFfFfFFFFFffe, 0x0000000000000000000000000000000000001Ac6, 0x00000000000000000000000000000000000001fa, 0x000000000000000000000000000000000000019B, 0x0000000000000000000000000000000000000000]`

Reply:
fixed